reporting_app/static/charts.js

```diff
@@ -0,0 +1,228 @@
+
+    var run_yield_data_months = new google.visualization.DataTable();
+    run_yield_data_months.addColumn('date', 'X');
+    run_yield_data_months.addColumn('number', 'Yield');
+    run_yield_data_months.addRows(by_month_yield_data);
+
+    var cumulative_run_yield_data_week = new google.visualization.DataTable();
+    cumulative_run_yield_data_week.addColumn('date', 'X');
+    cumulative_run_yield_data_week.addColumn('number', 'Yield');
+    cumulative_run_yield_data_week.addRows(by_week_yield_data_cumulative);
+
+    var cumulative_run_yield_data_month = new google.visualization.DataTable();
+    cumulative_run_yield_data_month.addColumn('date', 'X');
+    cumulative_run_yield_data_month.addColumn('number', 'Yield');
+    cumulative_run_yield_data_month.addRows(by_month_yield_data_cumulative);
+
+    // google charts options
+    var run_yield_options_weeks = {
+        title: 'Run yield per week',
+        hAxis: {title: 'Week / Year'},
+        vAxis: {title: 'Yield'}
+    };
+    var run_yield_options_months = {
+        title: 'Run yield per month',
+        hAxis: {title: 'Month / Year'},
+        vAxis: {title: 'Yield'}
+    };
+    var cumulative_run_yield_options_week = {
+        title: 'Cumulative run yield per week',
+        hAxis: {title: 'Week / Year'},
+        vAxis: {title: 'Cumulative Yield'}
+    };
+    var cumulative_run_yield_options_month = {
+        title: 'Cumulative run yield per month',
+        hAxis: {title: 'Month / Year'},
+        vAxis: {title: 'Cumulative Yield'}
+    };
+
+    // draw run charts
+    var yield_chart = new google.visualization.ColumnChart(document.getElementById('run_yield_by_date'));
+    var cumulative_yield_chart = new google.visualization.LineChart(document.getElementById('cumulative_run_yield_by_date'));
+
+    yield_chart.draw(run_yield_data_months, run_yield_options_months);
+    cumulative_yield_chart.draw(cumulative_run_yield_data_month, cumulative_run_yield_options_month);
+
+    var show_months_cumulative = document.getElementById("ShowCumulativeRunYieldMonths");
+    show_months_cumulative.onclick = function()
+    {
+        cumulative_yield_chart.draw(cumulative_run_yield_data_month, cumulative_run_yield_options_month);
+        yield_chart.draw(run_yield_data_months, run_yield_options_months);
+    }
+    var show_weeks_cumulative = document.getElementById("ShowCumulativeRunYieldWeeks");
+    show_weeks_cumulative.onclick = function()
+    {
+        cumulative_yield_chart.draw(cumulative_run_yield_data_week, cumulative_run_yield_options_week);
+        yield_chart.draw(run_yield_data_weeks, run_yield_options_weeks);
+    }
+}
+
+function unwind_samples_sequenced(sample_data){
+    var all_sample_data = [];
+    for (var e=0; e < sample_data.length; e++) {
+        d = sample_data[e];
+        run_ids = d['all_run_ids'].slice(0);
+        if (run_ids.length > 0){
+            run_ids.sort();
+            console.log(d['sample_id']);
+            console.log(run_ids)
```

Comment:
Don't need these print statements, as far as I can see. I'd try to print as little as possible, and use `console.warn` or `console.error` when stuff goes wrong.